web-ui/src/components/reviews/TeamReviews.jsx

```diff
@@ -592,105 +504,131 @@ const TeamReviews = ({ onBack, periodId }) => {
+          'Content-Type': 'application/json;charset=UTF-8',
+          'X-CSRF-Header': csrf
+        },
+        data: {
+          ...period,
+          reviewStatus: ReviewStatus.AWAITING_APPROVAL
         }
-      };
-
-      handleCloseNewRequest();
-      if (csrf && selectedMemberProfile && period) {
-        createNewRequest().then(res => {
-          if (res) {
-            window.snackDispatch({
-              type: UPDATE_TOAST,
-              payload: {
-                severity: 'success',
-                toast: 'Review request sent!'
-              }
-            });
-          }
+      });
+      if (res.error) throw new Error(res.error.message);
+      onBack();
+    } catch (err) {
+      console.error('TeamReviews.jsx deleteReviewer:', err);
+    }
+  };
+
+  const compareStrings = (s1, s2) => (s1 || '').localeCompare(s2 || '');
+
+  const sortMembers = members =>
+    members.sort((a, b) => {
+      let compare = compareStrings(a.lastName, b.lastName);
+      if (compare === 0) compare = compareStrings(a.firstName, b.firstName);
+      return compare;
+    });
+
+  const updateReviewers = async (member, reviewers) => {
+    const memberId = member.id;
+
+    let newAssignments = [...assignments];
+
+    // Remove all assignments for this member.
+    newAssignments = newAssignments.filter(a => a.revieweeId !== memberId);
+
+    // Add assignments for these reviewers if they don't already exist.
+    // All objects in the assignments array are for the current review period.
+    for (const reviewer of reviewers) {
+      const exists = newAssignments.some(
+        a => a.reviewerId === reviewer.id && a.revieweeId === memberId
+      );
+      if (!exists) {
+        newAssignments.push({
+          reviewPeriodId: periodId,
+          reviewerId: reviewer.id,
+          revieweeId: member.id
         });
       }
-    },
-    [
-      csrf,
-      dispatch,
-      handleCloseNewRequest,
-      period,
-      reviews,
-      selectedMemberProfile
-    ]
-  );
+    }
 
-  const createSecondary = teamMember =>
-    getReviewStatus(teamMember?.id) +
-    ', Self-review: ' +
-    getSelfReviewStatus(teamMember?.id);
+    try {
+      const res = await resolve({
+        method: 'POST',
+        url: `${reviewAssignmentsUrl}/${periodId}`,
+        data: newAssignments,
+        headers: {
+          'X-CSRF-Header': csrf,
+          Accept: 'application/json',
+          'Content-Type': 'application/json;charset=UTF-8'
+        }
+      });
+      if (res.error) throw new Error(res.error.message);
+      newAssignments = sortMembers(res.payload.data);
+      setAssignments(newAssignments);
+    } catch (err) {
+      console.error('TeamReviews.jsx updateTeamMembers:', err);
+    }
+  };
 
-  return (
-    <Root>
-      <Tooltip title="Back">
-        <Button onClick={onBack} aria-label="Back">
-          <ArrowBack /> Back
-        </Button>
-      </Tooltip>
+  const closeReviewerDialog = () => {
```

Comment:
This is called when the "X" button in the upper-left of the `MemberSelectorDialog` is clicked.